File: AndroidStudioProjects/MultiScreen/app/src/main/java/com/example/multiscreen/MainActivity.java
package com.example.multiscreen;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {
    EditText name;
    String nameText;
    public static final String EXTRA_NAME = "com.example.multiscreen.extra.name";
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button button = findViewById(R.id.button);


        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Toast.makeText(MainActivity.this,"This is the second activity",Toast.LENGTH_SHORT).show();//Toast
                Intent intent = new Intent(MainActivity.this , MainActivity2.class);
                name = findViewById(R.id.nameo);
                nameText = name.getText().toString();
                intent.putExtra("EXTRA_NAME",nameText);
                startActivity(intent);

            }
        });
    }



}

File: AndroidStudioProjects/Greetings/app/src/main/java/com/example/greetings/MainActivity.kt
package com.example.greetings

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import org.w3c.dom.Text

class MainActivity : AppCompatActivity() {

    lateinit var editText: EditText
    lateinit var button: Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        button = findViewById(R.id.click_me)
        button.setOnClickListener {

            editText = findViewById(R.id.EnterYourName)
            val name = editText.text
            Toast.makeText(this, "Good morning $name", Toast.LENGTH_SHORT).show()
        }

    }


}